### ISSUE TITLE
compares: check lhs and rhs types

### ISSUE THREAD

**Issue by Antonboom** [Antonboom/testifylint#205]
To protect from `"Elements should be the same type"` error

```go
func compareTwoValues(t TestingT, e1 interface{}, e2 interface{}, allowedComparesResults []CompareType, failMessage string, msgAndArgs ...interface{}) bool {
	if h, ok := t.(tHelper); ok {
		h.Helper()
	}

	e1Kind := reflect.ValueOf(e1).Kind()
	e2Kind := reflect.ValueOf(e2).Kind()
	if e1Kind != e2Kind {
		return Fail(t, "Elements should be the same type", msgAndArgs...)
	}
// ...
```
